File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/repository/MasterCodeRepository.java
package com.nivasafinance.features.master.codemaster.repository;

import com.nivasafinance.features.master.codemaster.entity.MasterCode;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface MasterCodeRepository extends JpaRepository<MasterCode, Long> {
    Optional<MasterCode> findByKey(String key);
    List<MasterCode> findByParentId(Long parentId);
    Page<MasterCode> findByParentId(Long parentId, Pageable pageable);
}

File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/repository/MasterCodeRepository.kt
package com.nivasafinance.features.master.codemaster.repository

import com.nivasafinance.features.master.codemaster.entity.MasterCode
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional
import java.util.UUID

@Repository
interface MasterCodeRepository : JpaRepository<MasterCode, UUID> {
    fun findByKey(key: String): Optional<MasterCode>
    fun findByParentId(parentId: UUID): List<MasterCode>
}
